app/assets/javascripts/initializers/initializeDrawerSliders.js

```diff
@@ -1,57 +1,50 @@
 function initializeDrawerSliders() {
-  if(!initializeSwipeGestures.called) {
-    swipeState = "middle";
+  if (!initializeSwipeGestures.called) {
+    swipeState = 'middle';
     initializeSwipeGestures();
   }
-  if (document.getElementById("on-page-nav-controls")){
-    if (document.getElementById("sidebar-bg-left")){
-      document.getElementById("sidebar-bg-left").onclick = function(){
-        swipeState = "middle";
-        slideSidebar("left","outOfView");
-      }
-    }
-    if (document.getElementById("sidebar-bg-right")){
-      document.getElementById("sidebar-bg-right").onclick = function(){
-        swipeState = "middle";
-        slideSidebar("right","outOfView");
-      }
-    }
 
-    if (document.getElementById("on-page-nav-butt-left")){
-      document.getElementById("on-page-nav-butt-left").onclick = function(){
-        swipeState = "left"
-        slideSidebar("left","intoView");
-      }
-    }
-    if (document.getElementById("on-page-nav-butt-right")){
-      document.getElementById("on-page-nav-butt-right").onclick = function(){
-        swipeState = "right"
-        slideSidebar("right","intoView");
+  if (document.getElementById('on-page-nav-controls')) {
+    const drawerSliders = [
+      { selector: 'sidebar-bg-left', swipeState: 'middle', side: 'left', view: 'outOfView', },
+      { selector: 'sidebar-bg-right', swipeState: 'middle', side: 'right', view: 'outOfView', },
+      { selector: 'on-page-nav-butt-left', swipeState: 'left', side: 'left', view: 'intoView', },
+      { selector: 'on-page-nav-butt-right', swipeState: 'right', side: 'right', view: 'intoView', },
+    ];
+
+    drawerSliders.forEach(drawerSliders => {
+      const element = document.getElementById(drawerSlider.selector);
+      if (element) {
+        element.onclick = function() {
```

Comment:
Instead of having 4 `click` events, you can add one `click` event on the element with the ID `on-page-nav-controls`. From there you can compare the `Event.target.id` with the `drawerSliders`. To avoid looping through the array, I'd suggest making it an object instead, e.g.

```
const drawerSliders = {
  'sidebar-bg-left': { swipeState: 'middle', side: 'left', view: 'outOfView' },
  'sidebar-bg-right': {
    swipeState: 'middle',
    side: 'right',
    view: 'outOfView',
  },
  'on-page-nav-butt-left': {
    swipeState: 'left',
    side: 'left',
    view: 'intoView',
  },
  'on-page-nav-butt-right': {
    swipeState: 'right',
    side: 'right',
    view: 'intoView',
  },
};
```

Reply:
I get the ideia of the `addEventListener`, but I don't see where/how exactly to implement this. :thinking:

Reply:
I'll post an example in a bit. Just in a meeting.

Reply:
I've made some testing here. I'll post it.
Using the array, it works fine:

`if (document.getElementById('on-page-nav-controls')) {
    const drawerSliders = [
      { selector: 'sidebar-bg-left', swipeState: 'middle', side: 'left', view: 'outOfView' },
      { selector: 'sidebar-bg-right', swipeState: 'middle', side: 'right', view: 'outOfView' },
      { selector: 'on-page-nav-butt-left', swipeState: 'left', side: 'left', view: 'intoView' },
      { selector: 'on-page-nav-butt-right', swipeState: 'right', side: 'right', view: 'intoView' },
    ];
     
    drawerSliders.forEach((drawerSlider) => {
      const element = document.getElementById(drawerSlider.selector);
      if (element) {
        element.onclick = function() {
          swipeState = drawerSlider.swipeState;
          slideSidebar(drawerSlider.side, drawerSlider.view);
        };
      }
    });
    body.addEventListener('click', () => {
      slideSidebar('right', 'outOfView');
      slideSidebar('left', 'outOfView');
    });
    listenForNarrowMenuClick();
  }
}`

But using as an object, the side bar doesn't close when I click elsewhere:

`   const element = document.getElementById('on-page-nav-controls')

  if (element) {
    const drawerSliders = {
      'sidebar-bg-left': { swipeState: 'middle', side: 'left', view: 'outOfView' },
      'sidebar-bg-right': { swipeState: 'middle', side: 'right', view: 'outOfView' },
      'on-page-nav-butt-left': { swipeState: 'left', side: 'left', view: 'intoView' },
      'on-page-nav-butt-right': { swipeState: 'right', side: 'right', view: 'intoView' },
    };

    element.addEventListener('click', (e) => {
      const drawerSlider = drawerSliders[e.target.id]
      swipeState = drawerSlider.swipeState;
      slideSidebar(drawerSlider.side, drawerSlider.view);
    });

    body.addEventListener('click', () => {
      slideSidebar('right', 'outOfView');
      slideSidebar('left', 'outOfView');
    });
    listenForNarrowMenuClick();
  }
}`

Any suggestions or observations?
Also, meanwhile I'm working on the others changes

Reply:
This is what I meant if using an object for drawer sliders, but the same could be done with the array. The only difference with the array is you'd need to loop through or do a `.find` to, get the right slider when comparing it to the clicked element's `className`. My assumption for this is there is only that one class name, which may not be correct. If there is more than one class name, looping through the array may make more sense to find the right drawer slider.

```javascript
const navControls = document.getElementById("on-page-nav-controls");

const drawerSliders = {
  'sidebar-bg-left': { swipeState: 'middle', side: 'left', view: 'outOfView' },
  'sidebar-bg-right': {.
    swipeState: 'middle',
    side: 'right',
    view: 'outOfView',
  },
  'on-page-nav-butt-left': {
    swipeState: 'left',
    side: 'left',
    view: 'intoView',
  },
  'on-page-nav-butt-right': {
    swipeState: 'right',
    side: 'right',
    view: 'intoView',
  },
};

navControls.addEventListener('click', e => {
  // e.target.className would be for example 'sidebar-bg-left'
  const drawerSlider = drawerSliders[e.target.className]

  if (!drawerSlider) {
    // a drawer slider was not clicked.
    return;
  }

  const { side, view } = drawerSlider;

  slideSidebar(side, view);

  // This appears to be a global as it is not set in this file
  swipeState = drawerSlider.swipeState;
});
```